### PR TITLE
fix: modifies UUID generation logic

### PR DESCRIPTION
**Pull request by matheusnascgomes**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/falso/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #10

## What is the new behavior?
It is now possible to have hundreds of often unique UUIDs, using a lightweight script that follows this UUID pattern as closely as possible.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/falso/src/tests/random-uuid.spec.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { randomUuid } from '../lib/random-uuid';
+describe('RandomUUIDTest', () => {
+  it('Should match UUID regex', () => {
+    const uuid = randomUuid();
+    const regex =
+      /^[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$/g;
+    expect(uuid.match(regex)).toBeTruthy();
+  });
+
+  it("Shouldn't duplicate element into a range of 10", () => {
```
Comment (theryansmee) [inline packages/falso/src/tests/random-uuid.spec.ts:10]: Sorry to be a pain but for consistency, I think this should use single quotes. (With a backslash to escape the apostrophe)

Comment (matheusnascgomes) [reply packages/falso/src/tests/random-uuid.spec.ts:10]: Hi @theryansmee, I agree and that's what I did, but Prettier turned it into double quotes. I will consider writing a positive sentence.

Comment (theryansmee) [reply packages/falso/src/tests/random-uuid.spec.ts]: Ah sorry, I don't had a lot of experience with prettier. Maybe `should not` to fudge it? ha

Comment (theryansmee) [reply packages/falso/src/tests/random-uuid.spec.ts]: @matheusnascgomes Ah sorry, I don't had a lot of experience with prettier. Maybe `should not` to fudge it? ha

File: packages/falso/src/lib/random-uuid.ts [hunk 2]
```diff
@@ -1,106 +1,31 @@
...
-    '8ca3de45-2a4b-4f85-9002-d75748971e02',
-    '3e4f747c-ad9f-4345-9c70-ed112dd60b34',
-    '319c243d-2908-4f48-8d00-bc7b2fd475f2',
-    '233d6d98-6d6a-4bb6-97f3-90a467fbd0a5',
-    '7591d3d9-0d36-4631-9257-5d5e03746e34',
-    'e982c2a1-37ea-4d4c-bf41-f9a3b033b10a',
-    'd86c9da0-9e66-46c0-b88b-b47f0a231494',
-    'a432864a-85f7-4202-9312-b34590c6fb63',
-  ]);
+  const xGroup1 = getGroupOfX(8);
```
Comment (NetanelBasal) [inline packages/falso/src/lib/random-uuid.ts:10]: Any reason you didn't use the simplest solution from here https://stackoverflow.com/questions/105034/how-to-create-a-guid-uuid?

File: packages/falso/src/tests/random-uuid.spec.ts (hunk 1 above)
Comment (geromegrignon) [reply packages/falso/src/tests/random-uuid.spec.ts]: With lint-staged on the project and given the prettier configuration used, the file should be prettified with single quotes.
As I have an upcoming PR for the project, I'll check if it works as expected.

File: packages/falso/src/lib/random-uuid.ts (hunk 2 above)
Comment (matheusnascgomes) [reply packages/falso/src/lib/random-uuid.ts:10]: It was mostly a personal decision, actually, to reduce third-party dependencies for such a simple need. But, I'm okay with use a library.

Comment (NetanelBasal) [reply packages/falso/src/lib/random-uuid.ts:10]: I don't mean using a library. I say that there are smaller solutions.

Comment (matheusnascgomes) [reply packages/falso/src/lib/random-uuid.ts]: Done. Also, I had to fix a test case due to `seedrandom` that was not installed in project's package.json

File: package.json [hunk 3]
```diff
@@ -12,7 +12,9 @@
     "packages/falso"
   ],
   "dependencies": {
-    "tslib": "2.0.0"
+    "seedrandom": "^3.0.5",
```
Comment (iamandrewluca) [inline package.json:15]: `seedrandom` already was added in another PR that is merged already, please sync branch with main

File: packages/falso/src/tests/random-uuid.spec.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { randomUuid } from '../lib/random-uuid';
+describe('RandomUUIDTest', () => {
```
Comment (iamandrewluca) [inline packages/falso/src/tests/random-uuid.spec.ts:2]:
```suggestion
describe('random-uuid', () => {
```

or 

```suggestion
describe('randomUuid', () => {
```

File: packages/falso/src/tests/random-uuid.spec.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  it('Should match UUID regex', () => {
```
Comment (NetanelBasal) [inline packages/falso/src/tests/random-uuid.spec.ts:3]:
```suggestion
  it('should match UUID regex', () => {
```

File: packages/falso/src/tests/random-uuid.spec.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    const uuid = randomUuid();
+    const regex =
+      /^[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$/g;
+    expect(uuid.match(regex)).toBeTruthy();
+  });
+
+  it('Should ensure no duplicate elements into a range of 10', () => {
```
Comment (NetanelBasal) [inline packages/falso/src/tests/random-uuid.spec.ts:10]:
```suggestion
  it('should ensure no duplicate elements into a range of 10', () => {
```

File: packages/falso/src/lib/random-uuid.ts [hunk 7]
```diff
@@ -1,106 +1,7 @@
...
-    'e982c2a1-37ea-4d4c-bf41-f9a3b033b10a',
-    'd86c9da0-9e66-46c0-b88b-b47f0a231494',
-    'a432864a-85f7-4202-9312-b34590c6fb63',
-  ]);
+import { v4 as uuidv4 } from 'uuid';
+/**
+ * Random valid UUID
+ */
+export function randomUuid(): string {
+  return uuidv4();
```
Comment (NetanelBasal) [inline packages/falso/src/lib/random-uuid.ts:6]: The library is not small. Can you check if it's tree-shakable and only pulls the code relevant to this function?

Comment (raisiqueira) [reply packages/falso/src/lib/random-uuid.ts:6]: I saw in bundlephobia, and they say that `uuid` is tree-shakable. 
image

Comment (NetanelBasal) [reply packages/falso/src/lib/random-uuid.ts:6]: Cool, so we don't need the specs.

File: packages/falso/src/tests/random-uuid.spec.ts (hunk 1 above)
Comment (geromegrignon) [reply packages/falso/src/tests/random-uuid.spec.ts]: I just checked and my staged spec file was prettified from double quotes to single quotes as expected.

You must have a local setting overriding the project's one.

File: packages/falso/src/lib/random-uuid.ts (hunk 7 above)
Comment (matheusnascgomes) [reply packages/falso/src/lib/random-uuid.ts:6]: As mentioned by @raisiqueira the lib is tree-shakable and the impact of an individual import is quite small. Note nx's report:

**WITH:**
Bundling falso...
  index.esm.js 824.486 KB
  index.cjs.js 828.891 KB
⚡ Done in 20.35s

**WITHOUT**
Bundling falso...
  index.esm.js 824.456 KB
  index.cjs.js 828.855 KB
⚡ Done in 11.60s

File: package.json (hunk 3 above)
Comment (matheusnascgomes) [reply package.json:16]: @iamandrewluca removing this from the global `package.json` breakes one test case (also in the pre-commit).
image

File: packages/falso/src/tests/random-uuid.spec.ts (hunk 1 above)
Comment (idanen) [reply packages/falso/src/tests/random-uuid.spec.ts]: For strings, Prettier will prefer the configured form, but will change to the othe when the string contains apostrophes or quotes.
For consistency, it's better to just use back ticks. Prettier won't change those

File: packages/falso/src/tests/random-uuid.spec.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import { randomUuid } from '../lib/random-uuid';
+describe('random-uuid', () => {
+  it('should match UUID regex', () => {
+    const uuid = randomUuid();
+    const regex =
+      /^[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$/g;
+    expect(uuid.match(regex)).toBeTruthy();
+  });
```
Comment (NetanelBasal) [inline packages/falso/src/tests/random-uuid.spec.ts:8]: please remove the specs.

File: package.json [hunk 9]
```diff
@@ -13,7 +13,9 @@
...
+    "tslib": "2.0.0",
+    "uuid": "^8.3.2"
```
Comment (NetanelBasal) [inline package.json:18]: move the uuid and the types to packages/falso/package.json

File: package.json [hunk 10]
```diff
@@ -13,7 +13,9 @@
     "packages/falso"
   ],
   "dependencies": {
-    "tslib": "2.0.0"
+    "seedrandom": "^3.0.5",
```
Comment (NetanelBasal) [inline package.json:16]: revert seedrandom changes, please.

File: packages/falso/package.json [hunk 11]
```diff
@@ -28,9 +28,11 @@
     "url": "https://github.com/ngneat/falso"
   },
   "dependencies": {
-    "seedrandom": "3.0.5"
+    "seedrandom": "3.0.5",
+    "uuid": "^8.3.2"
```
Comment (NetanelBasal) [inline packages/falso/package.json:32]: lock the version please

File: packages/falso/package.json [hunk 12]
```diff
@@ -28,9 +28,11 @@
...
   },
   "devDependencies": {
-    "@types/seedrandom": "3.0.1"
+    "@types/seedrandom": "3.0.1",
+    "@types/uuid": "^8.3.4"
```
Comment (NetanelBasal) [inline packages/falso/package.json:36]: lock the version please

**Review by NetanelBasal — APPROVED** [review]

File: packages/falso/package.json (hunk 11 above)
Comment (matheusnascgomes) [reply packages/falso/package.json]: done

**Pull request merged**